### PR TITLE
Goal Detail Page

### PR DESCRIPTION
**Pull request by Cwighty**
## **User description**
Could probably use another iteration to add some stuff

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Added new components for nutrient goals visualization including `DailyNutrientCarouselSelector`, `GoalsDetail`, `NutrientGoalChart`, `NutrientGoalRangeSummary`, `ProgressSummaryCard`, `TargetAmountChart`, and `NutrientProgress`.
- Introduced `useFetchGoalReportByNutrient` hook for fetching goal reports by nutrient.
- Updated `GoalsPage` to include `GoalsDetail` component.
- Customized scrollbar styles in theme configuration.
- Added `@mui/x-charts` to dependencies for chart visualization.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>12 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>DailyRecapCard.tsx</strong><dd><code>Update Import Path for NutrientProgress in DailyRecapCard</code></dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/dashboard/_components/Cards/DailyRecap/DailyRecapCard.tsx

- Updated import path for `NutrientProgress`.

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-2f46f0084fdbe7144d32ee0dcbc67b522c66246e66f82b37a1662bd3ec6ed51c">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DailyNutrientCarouselSelector.tsx</strong><dd><code>Add DailyNutrientCarouselSelector Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/DailyNutrientCarouselSelector.tsx

<li>Added new component <code>DailyNutrientCarouselSelector</code> for nutrient <br>selection.<br> <li> Utilizes <code>useFetchGoalReport</code> and <code>useCreateGoal</code> hooks for fetching and <br>creating goals.<br> <li> Implements nutrient selection UI with left and right navigation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-d2263f5f41ccde24fa73fc36ff823eba1a87a5636e6baea4a52eadbc3f1d2305">+136/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>GoalsDetail.tsx</strong><dd><code>Implement GoalsDetail Component for Nutrient Goals</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/GoalsDetail.tsx

<li>Introduced <code>GoalsDetail</code> component to display nutrient goals details.<br> <li> Uses <code>DailyNutrientCarouselSelector</code> for nutrient selection.<br> <li> Displays nutrient goal chart and summary.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-7cf7901774d7405aa6e1aacc0812c8297e2fe2c783cbc8850d2c50071da1a52f">+31/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NutrientGoalChart.tsx</strong><dd><code>Add NutrientGoalChart Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientGoalChart/NutrientGoalChart.tsx

<li>New <code>NutrientGoalChart</code> component for displaying nutrient consumption <br>chart.<br> <li> Utilizes <code>@mui/x-charts</code> for chart visualization.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-3c3853e512a5b5e2a8c05f14abcbba7422da69ab043952728ad8f08fc41a1d64">+70/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NutrientGoalRangeSummary.tsx</strong><dd><code>Add NutrientGoalRangeSummary Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientGoalChart/NutrientGoalRangeSummary.tsx

<li>New component <code>NutrientGoalRangeSummary</code> for displaying nutrient goal <br>progress summary.<br> <li> Includes date pickers for selecting date range and displays progress <br>cards.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-be7bd5f0faa0b2375c0df52b280a3e006df12ea4bd7d7260fccdc28be6502a72">+83/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ProgressSummaryCard.tsx</strong><dd><code>Implement ProgressSummaryCard Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientGoalChart/ProgressSummaryCard.tsx

<li>Introduced <code>ProgressSummaryCard</code> component for displaying progress <br>summary.<br> <li> Supports dynamic coloring based on progress status.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-db525bd407c3202ccdea8fc692686409d5ab98ef48f6ef2232f2acfcbcbad4a4">+40/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TargetAmountChart.tsx</strong><dd><code>Add TargetAmountChart Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientGoalChart/TargetAmountChart.tsx

<li>New <code>TargetAmountChart</code> component for displaying target amount lines on <br>charts.<br> <li> Supports custom and recommended target amounts.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-d5607e2cd237b835012d2679710300c2846b14903c010dc267c02e8caccc8c39">+61/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NutrientProgress.tsx</strong><dd><code>Add NutrientProgress Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientProgress.tsx

<li>New <code>NutrientProgress</code> component for displaying nutrient progress.<br> <li> Visualizes progress with linear progress bar and status icon.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-6b3fa4c6ba0a2c2eabb9e2dd84064341dc5c55b23251ec7c2fd0fd2b3633e616">+83/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>NutrientProgressProps.tsx</strong><dd><code>Define Props for NutrientProgress Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/_components/NutrientProgressProps.tsx

<li>Introduced <code>NutrientProgressProps</code> interface for <code>NutrientProgress</code> <br>component props.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-d6d921773a021aedae681f88e13928ba0d78770e0091cdb4492f4a84a3170e43">+14/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>goalHooks.ts</strong><dd><code>Add Hook for Fetching Goal Reports by Nutrient</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/goalHooks.ts

<li>Added <code>useFetchGoalReportByNutrient</code> hook for fetching goal reports by <br>nutrient.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-b282682a3d4f35d04fe210b01c42e6d5308921f7c1cefa8ed6f76bd87e346894">+18/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Update GoalsPage to Include GoalsDetail Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/app/(authorized)/goals/page.tsx

<li>Updated GoalsPage to include <code>GoalsDetail</code> component.<br> <li> Removed console log statement.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-edde738547b43767dcb762ef330789d82bc6029645da6036e0fd056fad62c290">+6/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ThemeContext.tsx</strong><dd><code>Customize Scrollbar Styles in Theme Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/src/context/ThemeContext.tsx

- Added custom scrollbar styles in theme configuration.

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-a0305ef3320f796c18876f592445a410e6b5a5a654982976d5919c117f75a6e6">+30/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug_fix
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>PatientContextPageContainer.tsx</strong><dd><code>Change Layout to Fixed Width in PatientContextPageContainer</code></dd></summary>
<hr>
      
nutrinova-ui/src/components/PatientContextPageContainer.tsx

<li>Changed <code>maxWidth</code> to <code>fixed</code> in <code>Container</code> component for consistent <br>layout.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-41de4f0e869029bcc2c279346ba268c682fa156801b124a4c2ba1443dd95030d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>next.config.js</strong><dd><code>Add @mui/x-charts to TranspilePackages in Next Config</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/next.config.js

<li>Added <code>@mui/x-charts</code> to <code>transpilePackages</code> for proper transpilation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-14a4ceb1424509f54673cdfb6ca1e71942ce562b2b09d8c7cdf03d9628921577">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies
</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>package-lock.json</strong><dd><code>Update Dependencies in package-lock.json</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/package-lock.json

- Updated package-lock.json with new dependencies.

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-9fbc79743a772a390f39c22c133f39899f6d4ef44d47f7b15af5c3ebf25209f2">+203/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>package.json</strong><dd><code>Add @mui/x-charts Dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
nutrinova-ui/package.json

- Added `@mui/x-charts` dependency for chart components.

</details>
    

  </td>
  <td><a href="https://github.com/Cwighty/nutrinova/pull/143/files#diff-dc0e3f9468f7bb5b6ec30ff8129cd53c7dbc9a72375f2d0d0ff381a9d705d78a">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: nutrinova-ui/src/app/(authorized)/goals/page.tsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 import { getServerSession } from "next-auth";
+import GoalsDetail from "./_components/GoalsDetail";
 
 export const metadata = {
   title: "Goals",
 };
 
 export default async function GoalsPage() {
   const session = await getServerSession();
+  console.log(session)
```
Comment (garv5014) [inline nutrinova-ui/src/app/(authorized)/goals/page.tsx:11]: Session console log

Comment (Cwighty) [reply nutrinova-ui/src/app/(authorized)/goals/page.tsx]: Youre right, im sorry

Comment (garv5014) [reply nutrinova-ui/src/app/(authorized)/goals/page.tsx]: No, está bien

**Review by garv5014 — APPROVED** [review]

File: nutrinova-ui/src/app/(authorized)/goals/page.tsx (hunk 1 above)
Comment (Cwighty) [reply nutrinova-ui/src/app/(authorized)/goals/page.tsx]: estoy bueno

**Pull request merged**